### ISSUE TITLE
bad input_ms.select_corr entries produce confusing errors

### ISSUE THREAD

**Issue by o-smirnov** [ratt-ru/QuartiCal#150]
I had a 2-corr MS, and I set ``input_ms.select_corr = [0,3]`` by mistake. This resulted in a rather baffling stack trace, as per below.

It is entirely user error of course, but a friendlier message would be nice.

```
#   File "/home/oms/.venv/stimela/lib/python3.8/site-packages/xarray/core/variable.py", line 1186, in isel                                         
#     return self[key]                                                                                                                             
#            │    └ (slice(None, None, None), slice(None, None, None), [0, 3])                                                                     
#            └ <xarray.Variable (row: 211792, chan: 1024, corr: 2)>                                                                                
#              dask.array<read~WEIGHT_SPECTRUM~[2,0,0]-1563189318_sdp_l0-A3528N_corr.ms...                                                         
#   File "/home/oms/.venv/stimela/lib/python3.8/site-packages/xarray/core/variable.py", line 779, in __getitem__                                   
#     data = as_indexable(self._data)[indexer]                                                                                                     
#            │            │    │      └ OuterIndexer((slice(None, None, None), slice(None, None, None), array([0, 3])))                            
#            │            │    └ <member '_data' of 'Variable' objects>                                                                            
#            │            └ <xarray.Variable (row: 211792, chan: 1024, corr: 2)>                                                                   
#            │              dask.array<read~WEIGHT_SPECTRUM~[2,0,0]-1563189318_sdp_l0-A3528N_corr.ms...                                            
#            └ <function as_indexable at 0x7fe5db79c280>                                                                                           
#   File "/home/oms/.venv/stimela/lib/python3.8/site-packages/xarray/core/indexing.py", line 1228, in __getitem__                                  
#     return self.array[key]                                                                                                                       
#            │    │     └ (slice(None, None, None), slice(None, None, None), array([0, 3]))                                                        
#            │    └ <member 'array' of 'DaskIndexingAdapter' objects>                                                                              
#            └ DaskIndexingAdapter(array=dask.array<read~WEIGHT_SPECTRUM~[2,0,0]-1563189318_sdp_l0-A3528N_corr.ms, shape=(211792, 1024, 2), ...    
#   File "/home/oms/.venv/stimela/lib/python3.8/site-packages/dask/array/core.py", line 1889, in __getitem__                                       
#     index2 = normalize_index(index, self.shape)                                                                                                  
#              │               │      │    └ <dask.utils.cached_property object at 0x7fe5e6968730>                                                 
#              │               │      └ dask.array<read~WEIGHT_SPECTRUM~[2,0,0]-1563189318_sdp_l0-A3528N_corr.ms, shape=(211792, 1024, 2),            
dtype=float32, chunksize=(...                                                                                                                          
#              │               └ (slice(None, None, None), slice(None, None, None), array([0, 3]))                                                 
#              └ <function normalize_index at 0x7fe5e63b7820>                                                                                      
#   File "/home/oms/.venv/stimela/lib/python3.8/site-packages/dask/array/slicing.py", line 917, in normalize_index                                 
#     check_index(axis, i, d)                                                                                                                      
#     │           │     │  └ 2                                                                                                                     
#     │           │     └ array([0, 3])                                                                                                            
#     │           └ 2                                                                                                                              
#     └ <function check_index at 0x7fe5e63b78b0>                                                                                                   
#   File "/home/oms/.venv/stimela/lib/python3.8/site-packages/dask/array/slicing.py", line 980, in check_index                                     
#     raise IndexError(                                                                                                                            
#                                                                                                                                                  
# IndexError: Index is out of bounds for axis 2 with size 2 
```

**Comment by JSKenyon** [comment]
Ah, noted. Will sort out a fix.

**Comment by JSKenyon** [comment]
Fixed by #164.
